core/src/main/java/org/apache/accumulo/core/security/ColumnVisibility.java

```diff
@@ -384,7 +384,8 @@ Node parse_(byte[] expression) {
             while (index < expression.length && expression[index] != '"') {
               if (expression[index] == '\\') {
                 index++;
-                if (expression[index] != '\\' && expression[index] != '"')
+                if (index < expression.length && expression[index] != '\\'
```

Comment:
This code may be relying on the array index out of bounds exception, not sure.  Depends on if it always expects something after.

Reply:
Well if it helps, the ColumnVisibilityParser is only used in the ColumnVisibility private validate method, which is only called in one of the ColumnVisibility constructors.